### PR TITLE
depot_tools: Add version 2024.2.29@50de66

### PR DESCRIPTION
**Pull request by PerikiyoXD**
As the title says.

Part of the v8 fix operation.

### CODE REVIEW

**Comment by PerikiyoXD** [comment]
### Important: https://github.com/xmake-mirror/depot_tools needs updating first

**Comment by PerikiyoXD** [comment]
# CI runner state

## Archlinux
https://github.com/xmake-io/xmake-repo/actions/runs/8090269975/job/22108649012?pr=3368

> error: @programdir/core/sandbox/modules/os.lua:273: ./update_depot_tools: line 148: goma_ctl: command not found

./update_depot_tools is ran, might not be desired. I need more insight on Linux. I think the `xmake.lua` might need tweaks for Linux/Unix-like environments.

## Windows
https://github.com/xmake-io/xmake-repo/actions/runs/8090269977/job/22108651498?pr=3368

> python3 --version
> Python 3.7.9
> ninja --version
> error: @programdir\core\sandbox\modules\os.lua:337: cannot exec(ninja --version), No such file or directory

Missing ninja binary

## Mac
https://github.com/xmake-io/xmake-repo/actions/runs/8090269980/job/22108650436?pr=3368

> error: @programdir/core/sandbox/modules/os.lua:273: ./update_depot_tools: line 148: goma_ctl: command not found

Same as Archilinux

File: packages/d/depot_tools/xmake.lua [hunk 1]
```diff
@@ -6,6 +6,11 @@ package("depot_tools")
     add_urls("https://github.com/xmake-mirror/depot_tools.git",
              "https://chromium.googlesource.com/chromium/tools/depot_tools.git")
     add_versions("2022.2.1", "8a6d00f116d6de9d5c4e92acb519fd0859c6449a")
+    add_versions("2024.2.29", "50de666ba40a4808daf9791fece3d8a43228a1de")
+
+    if is_plat("windows") then
+        add_deps("ninja") -- ninja is required on windows
```
Comment (waruqi) [inline packages/d/depot_tools/xmake.lua:12]: please remove comment

Comment (PerikiyoXD) [reply packages/d/depot_tools/xmake.lua]: Done already

**Comment by PerikiyoXD** [comment]
https://github.com/xmake-io/xmake-repo/blob/eccd98aa0c705a89128b749b873366bf98bcf56b/packages/d/depot_tools/xmake.lua#L29

envs are not being passed on Unix/Linux environments.

Tested locally

**Comment by waruqi** [comment]
> https://github.com/xmake-io/xmake-repo/blob/eccd98aa0c705a89128b749b873366bf98bcf56b/packages/d/depot_tools/xmake.lua#L29
> 
> envs are not being passed on Unix/Linux environments.
> 
> Although Windows envs are being properly set
> 
> Tested locally

it's only for proxy setting.

**Comment by PerikiyoXD** [comment]
`goma_ctl` is in the `depot_tools` directory
https://gist.githubusercontent.com/PerikiyoXD/fe2e1bc33a7fba641145237adf4a4fc0/raw/414200ee1b7a8ca0a88137e8f4b0529efd8d70fb/depot_tools.ls

**Comment by PerikiyoXD** [comment]
I couldn't make Linux work, caused by ninja (Which is integrated into depot_tools) will not work, just eats ram until VM exhaustion (16GB). More research must be done.

I'm okay just supporting windows for now

File: packages/d/depot_tools/xmake.lua [hunk 2]
```diff
@@ -6,6 +6,11 @@ package("depot_tools")
     add_urls("https://github.com/xmake-mirror/depot_tools.git",
              "https://chromium.googlesource.com/chromium/tools/depot_tools.git")
     add_versions("2022.2.1", "8a6d00f116d6de9d5c4e92acb519fd0859c6449a")
+    add_versions("2024.2.29", "50de666ba40a4808daf9791fece3d8a43228a1de")
+
+    if is_plat("windows") then
+        add_deps("ninja")
```
Comment (waruqi) [inline packages/d/depot_tools/xmake.lua:12]: depot_tools contains ninja

**Comment by waruqi** [comment]
> I couldn't make it work on Linux, caused by ninja (Which is integrated into depot_tools) eating ram until VM exhaustion (16GB). More research must be done.
> 
> I'm okay just supporting windows for now

but if we want to update this package in xmake-repo, we need to ensure all ci tests are passed.

**Pull request merged**